compiler/src/main/kotlin/motif/compiler/Dependencies.kt

```diff
@@ -78,5 +82,41 @@ class Dependencies private constructor(
 
             return Dependencies(typeSpec, typeName, methods)
         }
+
+        private fun getRequestersByType(sinks: Iterable<Sink>): Iterable<TypeAndRequesters> {
+            return sinks
+                    .groupBy { it.type }
+                    .mapValues { sinksByType ->
+                        val requesters = sinksByType.value.map { sink ->
+                            when (sink) {
+                                is FactoryMethodSink -> Requester(
+                                        sink.parameter.owner.qualifiedName,
+                                        inspectAndReturnMethodName(sink.parameter.owner, sink.parameter.method))
+                                is AccessMethodSink -> Requester(
+                                        sink.accessMethod.scope.qualifiedName,
+                                        sink.accessMethod.method.name)
+                            }
+                        }
+                        TypeAndRequesters(sinksByType.key, requesters)
+                    }
+                    .values
+        }
+
+        private fun getJavadocForType(requesters: List<Requester>): String {
+            val requestersHtmlItems = requesters.joinToString(separator = "\n") { requester ->
+                "<li>{@link ${requester.callerQualifiedName}#${requester.callerMethodName}}</li>"
+            }
+            return "<ul>Requested from:\n$requestersHtmlItems\n</ul>\n"
+        }
+
+        private fun inspectAndReturnMethodName(owner: IrClass, method: IrMethod): String {
+            if (!"<init>".contentEquals(method.name)) {
```

Comment:
Expose an `isConstructor` property in `IrMethod` and implement in `CompilerMethod` and `IntelliJMethod`:

**IrMethod**
```kotlin
val isConstructor: Boolean
```

**CompilerMethod**
```kotlin
override val isConstructor = element.kind == ElementKind.CONSTRUCTOR
```

**IntelliJMethod**
```kotlin
override val isConstructor = psiMethod.isConstructor
```

Reply:
On it!